server/src/internalClusterTest/java/org/opensearch/search/stats/ConcurrentSearchStatsIT.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.stats;
+
+import org.opensearch.action.admin.indices.stats.IndicesStatsRequestBuilder;
+import org.opensearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.index.IndexModule;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.indices.IndicesQueryCache;
+import org.opensearch.indices.IndicesService;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.script.Script;
+import org.opensearch.script.ScriptType;
+import org.opensearch.search.SearchTimeoutIT;
+import org.opensearch.test.InternalSettingsPlugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import static org.opensearch.index.query.QueryBuilders.scriptQuery;
+import static org.opensearch.search.SearchTimeoutIT.ScriptedTimeoutPlugin.SCRIPT_NAME;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, numDataNodes = 2, numClientNodes = 0)
+public class ConcurrentSearchStatsIT extends OpenSearchIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(SearchTimeoutIT.ScriptedTimeoutPlugin.class, InternalSettingsPlugin.class);
```

Comment:
We added `ScriptedTimeoutPlugin`  just to make sure the query would actually consume some time? May be you could just copy the class as `ScriptedDelayPlugin` since it is not really supposed to timeout (and it is just one class).

Reply:
Yeah I was seeing the query time occasionally come back as 0ms so this is just to make sure the test isn't flaky.

The `ScriptedTimeoutPlugin` class seems generically useful though, I can see it being very helpful to add tests for https://github.com/opensearch-project/OpenSearch/issues/9645. Additionally, all it does is add a sleep there's nothing timeout specific about this pluginScript, so how about if I just rename the existing `ScriptedTimeoutPlugin` class?

Reply:
It just looks a bit off to see `ScriptedTimeoutPlugin` from `SearchTimeoutIT` in the `ConcurrentSearchStatsIT` (those just look very unrelated). I think adding this plugin (few lines) into `ConcurrentSearchStatsIT` would make tests cleaner.

Reply:
@reta Alright made this change in the latest revision. Thanks!